### PR TITLE
Enables Authenticated GitHub API Access

### PR DESCRIPTION
**Pull request by ArquintL**
The functions to retrieve a GitHub release asset have been extended such that an optional token can be passed to them such that an authenticated API access will be performed. This increases the rate limit as only 60 requests per hour an IP address can be made unauthenticated.

In addition, this PR extends the tests to perform these API accesses as well as the asset downloads themselves authenticated using the runner's token.

Last but not least, `GitHubZipExtractor` further simplifies downloading of GitHub assets as it sets the correct download headers and only calculates the download URL when a download will actually happen

### CODE REVIEW

File: src/dependencies/GitHubReleaseAsset.ts [hunk 1]
```diff
@@ -72,6 +83,20 @@ export class GitHubReleaseAsset {
         return this.getAssetUrlFromRelease(taggedRelease, assetName);
     }
 
+    /**
+     * Returns an octokit instance that optionally uses the token for authentiction
+     * @param token personal access token, OAuth token, installation access token, or JSON Web Token for GitHub App authentication
+     */
+    private static getOctokit(token?: string): Octokit {
```
Comment (fpoli) [inline src/dependencies/GitHubReleaseAsset.ts:90]: Maybe `buildOctokit` reflects better what happen.

File: src/dependencies/GitHubZipExtractor.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * - the correct header for downloading a GitHub release asset is set
+ * - if a GitHub token is provided it is used to perform the download as an authenticated user
+ */
+export class GitHubZipExtractor implements DependencyInstaller {
+    private sequence: InstallerSequence | undefined;
+
+    constructor(
+        readonly remoteUrlFn: () => Promise<string>,
+        readonly folderName: string, // non optional as we need to determine whether a download is necessary
+        readonly token?: string
```
Comment (fpoli) [inline src/dependencies/GitHubZipExtractor.ts:18]: Can it be that these are `public` by default? If it is so you could add `private`.

File: src/test/runTest.ts [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
+            .option('token', {
+                description: 'GitHub access token that should be used for GitHub API calls. '
+                    + 'Use the "TOKEN" environment variable for CI as node logs the command incl. arguments',
+                type: 'string',
+            })
+            .help() // show help if `--help` is used
+            .argv;
+
+        let extensionTestsEnv;
+        if (argv.token || process.env["TOKEN"]) {
```
Comment (fpoli) [inline src/test/runTest.ts:17]: What about `GITHUB_TOKEN`? I'm thinking of scripts with more than one token.

**Review by fpoli — APPROVED** [review]
LGTM! I left some minor comments.

File: src/dependencies/GitHubZipExtractor.ts (hunk 2 above)
Comment (ArquintL) [reply src/dependencies/GitHubZipExtractor.ts:18]: You're right, `public` is the default

**Comment by ArquintL** [comment]
Thanks for the review! I've implemented your suggestions

**Pull request merged**